radixdlt-engine/src/main/java/com/radixdlt/engine/RadixEngine.java

```diff
@@ -435,30 +438,33 @@ private TxBuilder construct(TxBuilderExecutable executable, Set<SubstateId> avoi
     }
   }
 
+  // This method is used only in tests
   public TxBuilder construct(TxAction action) throws TxBuilderException {
     return construct(TxnConstructionRequest.create().action(action));
   }
 
   public TxBuilder construct(TxnConstructionRequest request) throws TxBuilderException {
     var feePayer = request.getFeePayer();
+
     if (feePayer.isPresent()) {
       return constructWithFees(request, feePayer.get());
-    } else {
-      return construct(
-          txBuilder -> {
-            if (request.isDisableResourceAllocAndDestroy()) {
-              txBuilder.toLowLevelBuilder().disableResourceAllocAndDestroy();
-            }
-            for (var action : request.getActions()) {
-              this.actionConstructors.construct(action, txBuilder);
-            }
-            var msg = request.getMsg();
-            if (msg.isPresent()) {
-              txBuilder.message(msg.get());
-            }
-          },
-          request.getSubstatesToAvoid());
     }
+
+    return construct(
+        txBuilder -> {
+          if (request.isDisableResourceAllocAndDestroy()) {
+            txBuilder.toLowLevelBuilder().disableResourceAllocAndDestroy();
+          }
+          for (var action : request.getActions()) {
+            this.actionConstructors.construct(action, txBuilder);
+          }
+
+          var msg = request.getMsg();
+          if (msg.isPresent()) {
+            txBuilder.message(msg.get(), maxMessageLen);
```

Comment:
This seems like antipattern here, you already know messageLen so why pass this in? Instead, should pass maxMessageLen into `TxBuilder` constructor

Reply:
Fixed